### PR TITLE
P11 develop debug

### PR DESCRIPTION
**Pull request by agruzinov**
Debug after upgrading to the latest development.

### CODE REVIEW

File: mxcubecore/HardwareObjects/DESY/P11SampleChanger.py [hunk 1]
```diff
@@ -486,3 +486,19 @@ def _update_selection(self):
                 s._set_loaded(False)
 
         self._set_selected_sample(None)
+    
+    def _set_loaded_sample(self, sample):
```
Comment (beteva) [inline mxcubecore/HardwareObjects/DESY/P11SampleChanger.py:490]: @agruzinov, is defining this method solves the recurrence problem you've mentioned?

File: mxcubecore/HardwareObjects/DESY/P11Collect.py [hunk 2]
```diff
@@ -144,6 +136,8 @@ def data_collection_hook(self):
 
         osc_pars = self.current_dc_parameters["oscillation_sequence"][0]
         file_info = self.current_dc_parameters["fileinfo"]
+        osc_pars["kappa"] = 0
+        osc_pars["kappa_phi"] = 0
```
Comment (beteva) [inline mxcubecore/HardwareObjects/DESY/P11Collect.py:276]: Do you have minikappa head on your diffarctometer?

Comment (agruzinov) [reply mxcubecore/HardwareObjects/DESY/P11Collect.py:276]: No, this is just not to have none values, just in case. Does it make sense?

Comment (beteva) [reply mxcubecore/HardwareObjects/DESY/P11Collect.py:276]: It is OK if you would never use minikappa.

File: mxcubecore/HardwareObjects/DESY/P11SampleChanger.py (hunk 1 above)
Comment (agruzinov) [reply mxcubecore/HardwareObjects/DESY/P11SampleChanger.py:490]: Yes, it solves the "loading failed" problem.

**Comment by agruzinov** [comment]
Funny enough after synchronising with the latest develop on github webpage, linting test is failing now, although I did not change a contenent.

File: mxcubecore/HardwareObjects/DESY/P11Collect.py [hunk 3]
```diff
@@ -330,33 +444,29 @@ def collect_characterisation(
...
             stop_angle = start_at + img_range * 1.0
 
-            logging.info("collecting image %s, angle %f" % (img_no, start_at))
+            self.log.debug("collecting image %s, angle %f" % (img_no, start_at))
 
+            # Keep it here for now. It is not clear if it is needed.
             if start_at >= stop_angle:
-                init_pos = start_at
-
+                init_pos = start_at  # - self.acq_speed * self.turnback_time
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/DESY/P11Collect.py]: Currently both branches of the 'if' statement are identical - and anyway init_pos is not used so both do nothing. Harmless, but maybe a clean-up or comment-out?

**Review by rhfogh — APPROVED** [review]
Fine with me

File: mxcubecore/HardwareObjects/DESY/P11Collect.py (hunk 3 above)
Comment (agruzinov) [reply mxcubecore/HardwareObjects/DESY/P11Collect.py]: Did it, thanks! It is still not clear if this is an issue or not.

File: mxcubecore/HardwareObjects/DESY/P11NanoDiff.py [hunk 4]
```diff
@@ -333,7 +333,7 @@ def manual_centring(self, phi_range=120, n_points=3):
             self.user_clicked_event = AsyncResult()
             x, y = self.user_clicked_event.get()
             if click < 2:
-                phi_mot.set_value_relative(phi_range)
+                phi_mot.set_value_relative(phi_range, timeout=5)
```
Comment (beteva) [inline mxcubecore/HardwareObjects/DESY/P11NanoDiff.py]: If your phi_mot  inherits from AbstractMotor, than only means that you will wait max 5s for the motor to move an will raise TimeoutError if the movement takes longer. It does not give you an extra time for the clicking.
Maybe your problem was that by default the timeout=0, meaning, do not wait for the movement to finish.

File: mxcubecore/HardwareObjects/DESY/P11Collect.py [hunk 5]
```diff
@@ -470,18 +577,19 @@ def add_h5_info(self, h5file):
...
+                #     1.0,
+                #     0.0,
+                #     0.0,
+                # ]
+                # h5fd["entry/instrument/detector/module/fast_pixel_direction"].attrs[
+                #     "vector"
+                # ] = [1.0, 0.0, 0.0]
+                # h5fd["entry/instrument/detector/module/slow_pixel_direction"].attrs[
+                #     "vector"
+                # ] = [0.0, 1.0, 0.0]
```
Comment (beteva) [inline mxcubecore/HardwareObjects/DESY/P11Collect.py:592]: This represents the direction of the diffractometer head in your metadata image. If your detector is Eiger and you use the dectris filewriter, these values are set by default  as horizontal orientation, so you need to change to something else if your diffractometer is not the same.

File: mxcubecore/HardwareObjects/DESY/P11NanoDiff.py (hunk 4 above)
Comment (agruzinov) [reply mxcubecore/HardwareObjects/DESY/P11NanoDiff.py]: Thanks for the comment! What would be the best way to mitigate it? I do not see any "wait" option there.

File: mxcubecore/HardwareObjects/DESY/P11Collect.py (hunk 5 above)
Comment (agruzinov) [reply mxcubecore/HardwareObjects/DESY/P11Collect.py:592]: Thanks!

File: mxcubecore/HardwareObjects/DESY/P11NanoDiff.py (hunk 4 above)
Comment (beteva) [reply mxcubecore/HardwareObjects/DESY/P11NanoDiff.py]: timeout=None waits forever. But you can as well put some big value, like 2 min (timeout=60). This is only security - the command will return as soon as the movement has finished. 

You may want to have a look at the AbstractMotor.py. It says:
` timeout (float): optional - timeout [s],
                             If timeout == 0: return at once and do not wait;
                             if timeout is None: wait forever
`

Comment (agruzinov) [reply mxcubecore/HardwareObjects/DESY/P11NanoDiff.py]: OK, thanks for the tip. Cheers!

**Pull request merged**